core/domain/stats_jobs_one_off.py

```diff
@@ -127,3 +143,2369 @@ def reduce(key, stringified_values):
+                    assumptions[0:-1], operators),
+                AnswerMigrationJob._display_expression_helper(
+                    assumptions[-1], operators, 0))
+
+        target_string = AnswerMigrationJob._display_expression_helper(
+            results[0], operators, 0)
+
+        # The resulting answer is only 'Correct' if it is not a default answer,
+        # since if it were correct then it would have been matched to the
+        # 'Correct' rule type.
+        correct = rule_str != cls._DEFAULT_RULESPEC_STR
+        if correct:
+            return cls._normalize_raw_answer_object(objects.CheckedProof, {
+                'assumptions_string': assumptions_string,
+                'target_string': target_string,
+                'proof_string': answer_str,
+                'correct': True
+            }, answer_str)
+        else:
+            return cls._normalize_raw_answer_object(objects.CheckedProof, {
+                'assumptions_string': assumptions_string,
+                'target_string': target_string,
+                'proof_string': answer_str,
+                'correct': False,
+                'error_category': '',
+                'error_code': '',
+                'error_message': '',
+                'error_line_number': -1
+            }, answer_str)
+
+    @classmethod
+    def _cb_reconstitute_math_expression_input(
+            cls, state, answer_group_index, rule_spec, rule_str, answer_str):
+        if rule_str != cls._DEFAULT_RULESPEC_STR:
+            if rule_spec.rule_type != 'IsMathematicallyEquivalentTo':
+                return (
+                    None,
+                    'Cannot reconstitute MathExpressionInput object without an '
+                    'IsMathematicallyEquivalentTo rule.')
+        math_expression_dict = ast.literal_eval(answer_str)
+        if not isinstance(math_expression_dict, dict):
+            return (
+                None,
+                'Bad answer string in MathExpressionInput '
+                'IsMathematicallyEquivalentTo rule.')
+        return cls._normalize_raw_answer_object(
+            objects.MathExpression, math_expression_dict, answer_str)
+
+    @classmethod
+    def _cb_reconstitute_multiple_choice_input(
+            cls, state, answer_group_index, rule_spec, rule_str, answer_str):
+        # The Jinja representation for NonnegativeInt answer strings is:
+        #   {{ choices[answer|int] }}
+        interaction = state.interaction
+        if rule_str == cls._DEFAULT_RULESPEC_STR or (
+                rule_spec.rule_type == 'Equals'):
+            customization_args = interaction.customization_args
+            choices = customization_args['choices']['value']
+
+            # An answer group index is only available for non-default rules.
+            if rule_str != cls._DEFAULT_RULESPEC_STR:
+                outcome = interaction.answer_groups[answer_group_index].outcome
+                temporary_special_param = next(
```

Comment:
That is a smell, especially the lack of documented-ness. Does the special parameter get removed when the answer is placed into the target datastore?

I would still be fine with not storing answers that do not have a subject of 'answer', by the way, especially if storing them results in harder-to-maintain code.